### ISSUE TITLE
Fix quests [RFC]

### ISSUE THREAD

**Issue by 0lafe** [GTNewHorizons/GT-New-Horizons-Modpack#6590]
#### Which modpack version are you using?
2.0.9

#### What do you suggest instead/what changes do you propose?

Quests atm are in a really weird spot, which causes a handful of issues. Non completable quests seem to pop up every version, and no one really enjoys crafting tasks. I have a solution to all these problems and more, while making quests more user friendly and promote new player growth. 

We should remove all rewards on most quests, and change them to check box tasks. Everything in a tier should open up at once, to give you perspective. 

The one thing I'm somewhat worried about is tiering up. I'm not opposed to having maybe one crafting task at the end of each tier, that gives some lootbags/coins, and unlocks the next tier. 

Basically makes the questbook an in-game wiki, instead of a weird focus point on "progression" with many kinks.

**Comment by yukieiji** [comment]
> We should remove all rewards on most quests, and change them to check box tasks.

Can it be called a quest?
I think some players enjoy the current questing system, so why not use the [In-Game wiki](https://www.curseforge.com/minecraft/mc-mods/in-game-wiki-mod) mod to create an in-game wiki?

**Comment by Prometheus0000** [comment]
Yukieji, you seem to be both underestimating the amount of work that would take, and the number of people who would do so. Plus that doesn't tell you how to progress, just what something does.

- I'm fine with changing crafting tasks to retrieval. 'learning the recipes' as dream says doesn't mean anything, people look them up in NEI, not memorize them. And if you did memorize them, that wouldn't help you either. Not to mention that almost all of them aren't findable in the world anyway. I'd also be fine changing just the ones you find in-world (excluding infernal drops!). See #6588.
- For quest rewards I'm not really sure, but if we could add a mode to BQ3 that didn't give rewards that would be useful for some people. You also need to keep in mind that this would make the coins tab useless, and make some things harder, like getting maple saplings, silverwood saplings, one of the hives, and spawning the TF bosses again. Those quests are there for a reason.
- Opening everything in a tier at once would lack any clear sense of what you had to do to progress, like that you need a better cutting machine to cut wafers, or that you need x to do y over and over. If you want to mention specific instances where making things optional would help, make a ticket.
- Non completeable quests are mostly because it was default nbtignore off before, but when it was changed, it didn't affect existing quests, and they didn't all have it set correctly.

**Comment by 0lafe** [comment]
The coins can be kept through giving progress on a few non check box tabs for tiering up. Give like a bunch of the tier you're going into, and let you use them throughout as you wish. 

The progression would be the same. All stuff leads into other things. It would lack everything we already lack, and still have everything we currently have. The same progression routes would still exist, so I don't think it would really lack continuity.

**Comment by Prometheus0000** [comment]
?

> Everything in a tier should open up at once, to give you perspective.

> The same progression routes would still exist


Wouldn't this be one quest in the middle, and 50+ connected to it? And maybe one off to the side to unlock the next tier? Or just 50 quests opened up with no indication of what to do? That's not progression.

**Comment by dvdmandt** [comment]
What if we made an addon/change/whatever to BQ and/or SQP to allow something like optional tasks/steps? That way, we could keep the current steps as explanations and only require the last step (which should probably be retrievals), yet make quests a lot less painful to progress on many cases. And/or some form of "Skip" (for some quests?) that can allow you move forward in the questline without getting the rewards?

**Comment by Prometheus0000** [comment]
I dunno, depends on what dream wants I guess.

**Comment by GTNH-Colen** [comment]
> I'm not opposed to having maybe one crafting task at the end of each tier, that gives some lootbags/coins, and unlocks the next tier.

No, please stop trying to spread the cancerous growth that is crafting quests.

**Comment by yukieiji** [comment]
@Prometheus0000 I don't think there's much difference between the job of removing almost all quest rewards and replacing them with checkbox tasks and the job of creating a new wiki with the In-game wiki mod.
There wouldn't be much difference between the job of changing 1000+ quests and the job of describing the details about 16 tiers.

**Comment by Prometheus0000** [comment]
I would refuse to do either of them even if you convinced dream, since it would be a _gigantic_ pain in the ass.

And do you not understand that copy-pasting text from wikis _alone_ would take solid weeks if not months? Not to mention that they're unreliable, so you can't do that for a bunch of it too, since they've been altered for the pack. There are thousands and thousands or even hundreds of thousands (I'm not really sure, but it's a lot) of items in the pack, even if you cut down on repeats with just the material changed, it's a huge amount of stuff. Plus with the additional text and images added, I image that load times and lag would increase too, though I'm not really sure.

**Comment by yukieiji** [comment]
I think if you were to change 1000+ quests it would take a few months.

All we need is a wiki to progress the tier, and you won't need to copy the entire wiki.(although there will still be more data, so load times will increase slightly).

**Comment by Prometheus0000** [comment]
Then do the wiki thing yourself? Again, not doing either.

**Comment by yukieiji** [comment]
I won't list anything on the In-Game wiki other than what's necessary for progression, it's just to help quest.
As an example, in the LV tab, I don't list anything other than the following(0lafe's needs may be more than this, but)
・In the next Tier, we need to craft a circuit for MV and Basic Circuit Assembler(Images of recipe).
・You'll need an emitter and assembler to craft them.
・Crafting (e.g. Plate) in LV, which was inefficient in previous Tier, is now more efficient with machines!
・If you're having trouble progressing, take a look at the quests.

**Comment by Prometheus0000** [comment]
That's not a wiki, that's just a list of items to do. You need to explain the in-depth detail of how the block works, and everything related to it, or it's not a wiki. What would be the point of this, when we have the quests that go into so much more detail? You're just compressing the quests into a single line each that explains nothing. This is a wiki page: https://ftb.gamepedia.com/Blood_Altar. Although it's still lacking in my opinion.

**Comment by yukieiji** [comment]
This is briefly written in one line, but if I'm writing this as a In-Game wiki, I'll explain it properly.

**Comment by NoX-programer** [comment]
I vote to improve BetterQuesting if its functionality is insufficient. The reason is not to spread progression info (quests), keep it all in one place. And I vote against adding mods where it is not absolutely necessary.
